graphql/src/main/kotlin/com/trib3/graphql/resources/GraphQLResource.kt

```diff
@@ -46,7 +48,8 @@ open class GraphQLResource
 @Inject constructor(
     private val graphQL: GraphQL,
     private val graphQLConfig: GraphQLConfig,
-    private val creatorFactory: GraphQLContextWebSocketCreatorFactory
+    private val creatorFactory: GraphQLContextWebSocketCreatorFactory,
+    @Nullable val dataLoaderRegistryFactory: DataLoaderRegistryFactory? = null
```

Comment:
Why both the `?` and the `@Nullable`

Reply:
? is for kotlin, @Nullable is for guice ( https://github.com/google/guice/wiki/UseNullable )

not that we still need to explicitly bind the object, can't just leave it unbound and get a null value.  so using an OptionalBinder in the default graphql module that defaults to null and apps can override.